### ISSUE TITLE
Clicking becomes restricted to one window

### ISSUE THREAD

**Issue by synackd** [swaywm/sway#5922]
### Sway Version

```
$ swaymsg -t get_version
sway version 1.5.1
```

### Debug Log

Without `xwayland disable` in config file: https://pastebin.com/4ZhsFbMq
With `xwayland disable` in config file: https://pastebin.com/jbCcQTMa

Between running `/bin/sway -d 2>~/sway.log` and exiting sway, I:

1. Open `$term` in workspace 1 (using `$mod + Enter`) and `cat` an arbitrary file (to fill the terminal with text).
2. Repeat 1 in workspace 2.
3. Open `$term` in workspace 3 and run `sudo libinput debug-events` with the `--device` flag for the following devices in order to ensure that events are still being captured when the problem occurs:
    - The external USB mouse
    - The trackpad
    - The trackpoint (which also includes the trackpad's click 

### Configuration File

The only modifications to the default config file I made were:

- Adding `xwayland disable` to test with/without xwayland (see above)
- Setting `$term` to `kitty` instead of the default `alacritty`

### Stack Trace

Not applicable since sway does not crash.

### Description

I am running Void Linux with kernel version 5.9.16_1 on a ThinkPad X220t.

At first, I thought that this problem was an instance of #4261 but after more detailed troubleshooting, found it to be different.

After

- a seemingly-arbitrary amount of time, or
- the machine resumes from suspend,

clicking _any_ mouse (trackpad, USB mouse, etc.) only works in one particular window. The cursor can still move around on any workspace, but clicking is limited to a particular window in a particular workspace. During this time, the status bar is unable to be clicked (tested using [Waybar](https://github.com/Alexays/Waybar)) as well. To remedy this, I have found that suspending/resuming _sometimes_ restores clicking ability to all windows, but quitting and restarting sway _always_ does.

Mouse/Trackpad events, including clicks, are still being registered by `libinput`, but it seems that no matter which window or workspace one is on, sway thinks that this one window is in focus, despite shifting focus away from this window. The keyboard still works perfectly, and it is the only way one can switch workspaces/windows when this problem occurs.

Interestingly, similar to #4261, clicking still works when clicking "Exit" in the sway nagbar to exit sway. It also may be worth noting that when I enable two-finger scrolling in the sway config, it does not work when sway starts until I switch to another TTY and switch back to the TTY sway is on.

To reproduce this problem, perform the steps outlined in [**Debug Log**](#debug-log) above and then:

4. Wait some time or, more reliably, suspend the machine, wait, and resume.
    - In my case, I use Void's [`zzz(8)`](https://man.voidlinux.org/zzz.8) utility.
5. Select some text in the `$term` window in workspace 1, then deselect it to verify that clicking works in this window.
6. Switch to workspace 2 and try to select the text in that `$term` window.
    - If the problem is present, one will not see anything be selected.
7. Switch back to workspace 1 and see that the corresponding section of text is selected on this `$term` window.

**Comment by synackd** [comment]
Not sure if it has anything to do with lines 842-843 of the debug log:
```
OSError: Failed to load the function ca_context_create with error: Error loading shared library libcanberra.so: No such file or directory
[366 12:26:09.975481] [glfw error 65549]: Wayland: Window attention request not implemented yet
```
`libcanberra` is installed but the `libcanberra.so` file doesn't seem to be present:
```$ locate libcanberra.so
/usr/lib/libcanberra.so.0
/usr/lib/libcanberra.so.0.2.5
```

Not sure why a sound event library would cause clicking problems...

The second line seems more relevant.

**Comment by synackd** [comment]
This behavior also occurs after resuming from a simple call to
```
$ echo mem | sudo tee /sys/power/state
```

**Comment by synackd** [comment]
The problem does not seem to occur when the laptop is plugged into external monitors, mouse, and keyboard via the dock.

**Comment by meddion** [comment]
@synackd I'm experiencing the very behavior you had described; is there any progress on the troubleshooting?

**Comment by synackd** [comment]
Nothing irregular in the logs as far as I can tell, though I find that:
- After closing the window that the clicking is restricted to, the clicking restriction occurs in _another_ window.
- This problem does not occur when using an external mouse/keyboard, but as soon as I try to drag the mouse using the trackpad, without clicking, the text in the window beneath is highlighted (as if I'm clicking while dragging) and the clicking restriction occurs.

@meddion What are you running when this occurs for you?

**Comment by synackd** [comment]
Also, using the trackpoint avoids this problem, it seems.

**Comment by meddion** [comment]
@synackd seems like it doesn't matter what software I'm running, whether it's through Xwayland or directly. 
I had found two temporary workarounds for this: 
- The first is to suspend my system (not hibernating) and restore it, after this everything works as expected for a few minutes.
- The second is to toggle the full mode on the problematic window through sway, then in my case, it will allow the mouse to escape it and be captured & locked inside another window (and so on if your continue). 

I am using an external USB mouse only since my touchpad has died a few weeks ago. I am beginning to suspect the latter to be somehow involved, I will later try to go through all of my input devices and analyze them to see if something weird is happening without my consent.

**Comment by meddion** [comment]
Indeed, my touchpad was causing the problem, or rather the accompanying mouse buttons. Disabling both these devices with the following entries in the sway config seems to eliminate the issue:
```
input type:touchpad events disabled
input "2:1:PS/2_Generic_Mouse" events disabled
```
Furthermore, it would be a good idea to disable these devices completely with udev.

**Comment by synackd** [comment]
I would prefer not to have to disable my touchpad to use sway, though.

I thought it might be tapping being enabled that was causing the issue, so I tried disabling it through
```
input type:touchpad tap disable
```
though to no avail.

**Comment by NatKarmios** [comment]
Apologies for bringing up such an old issue, but did you find out anything more on this @synackd? Interestingly, I'm encountering more or less the exact same issue on both Hyprland and Niri (have yet to try Sway).

**Comment by rhysperry111** [comment]
Can confirm I've seen this pop up once or twice recently - and toggling fullscreen on the application that's "stealing" mouse focus seems to fix things.

In my case Discord (through Xwayland) was stealing the mouse from everything else.

**Comment by NatKarmios** [comment]
My specific case is a bit more strange; the issue was my laptop's integrated keyboard occasionally sending a `BTN_0` `POINTER_BUTTON` event, which those compositors recognised as starting a click-and-drag, messing up subsequent mouse input.
I've currently fixed it with a local libinput quirk override (for which I've raised an issue [here](https://gitlab.freedesktop.org/libinput/libinput/-/issues/1157)).

**Comment by synackd** [comment]
I tried this again today using the trackpad and had the same issue, except this time I couldn't click on _anything_ afterward, no matter the workspace.

Logs (with xwayland): https://pastebin.com/6E37LTyJ

Linux 6.12.25_1
Sway 1.10
wlroots 0.18.1
